Development/src/pages/devices/ChannelMappingMatrix.js

```diff
@@ -36,12 +37,55 @@ import { useJSONSetting } from '../../settings';
 import labelize from '../../components/labelize';
 import { getFilteredInputs, getFilteredOutputs } from './FilterMatrix';
 
-const StyledTableCell = withStyles({
+// passing variant="head" doesn't seem to work inside TableBody
+const TableHeadCell = props => <TableCell component="th" {...props} />;
+
+const mappingCellStyle = theme => ({
+    textAlign: 'center',
+    padding: '1px 2px',
+    border: `solid 1px ${
+        theme.palette.type === 'dark' ? '#515151' : '#e0e0e0'
+    }`,
+});
+
+const mappingHeadStyle = theme => ({
+    backgroundColor: theme.palette.type === 'dark' ? '#212121' : '#f5f5f5',
+});
+
+const MappingCell = withStyles(theme => ({
+    root: mappingCellStyle(theme),
+}))(TableCell);
+
+// for column and row headings
+const MappingHeadCell = withStyles(theme => ({
     root: {
-        padding: 2,
-        border: '1px solid rgba(224, 224, 224, 1)',
+        ...mappingCellStyle(theme),
+        ...mappingHeadStyle(theme),
     },
-})(TableCell);
+}))(TableHeadCell);
+
+const MappingCornerCell = withStyles(theme => ({
+    root: {
+        ...mappingCellStyle(theme),
+        borderLeft: 0,
+        borderTop: 0,
+    },
+}))(TableHeadCell);
+
+// Midline Horizontal Ellipsis for when columns have been collapsed
+const HorizontalEllipsisButton = props => (
+    <IconButton size="small" children={'\u22ef'} {...props} />
+);
+
+// Vertical Ellipsis for when rows have been collapsed
+const VerticalEllipsisButton = props => (
+    <IconButton size="small" disabled children={'\u22ee'} {...props} />
+);
```

Comment:
We discussed that `⋮` may look too much like a menu icon...

(https://twitter.com/lukew/status/591296890030915585)

The checked and unchecked states of disabled `MappingButton`s are also quite hard to distinguish, so we may want to revisit the matrix presentation after user feedback.

Reply:
@rhastie, I pushed a commit to fade-back the unchecked icons and ellipses. I think this simple change improves the UX considerably!

Reply:
Yes - That is looking better... Is the "ellipse-tick icon" at 50% transparency (probably wrong term!)? Could we also slightly fade-forward the "ellipse-tick icon" a little bit so it was half-way between 0 and 100% transparency? This means we can then still use the much darker 0% transparency for the "staged map" screen to show an "active change" etc...

Reply:
The empty circle and the three-dots icons are now all faded back by applying 0.3x opacity. On top of that, all the icons in the Active Map, including those ones _and_ the ticked circle, are flagged as 'disabled' for which Material UI seems to apply 0.26x opacity by default. We want the 'disabled' state in the Active Map view to be significantly different to the 'enabled' state in the Staged Map, but 0.4x might be a compromise...

Reply:
Sounds a plan... IMHO we can dial the opacity back up if needed when we get to the Staged Map